addons/purchase/static/src/js/purchase_portal_sidebar.js

```diff
@@ -134,4 +139,12 @@ publicWidget.registry.PurchasePortalSidebar = PortalSidebar.extend({
         });
         return rawText.join(" ");
     },
+    _onDecline: function(ev){
+        let orderId = parseInt(ev.currentTarget.dataset.orderId)
+        this.orm.call("purchase.order","decline",[orderId])
+    },
+    _onAccept: function(ev){
+        let orderId = parseInt(ev.currentTarget.dataset.orderId)
+        this.orm.call("purchase.order","accept",[orderId])
+    }
```

Comment:
Shouldn't the view be refreshed ? I can click twice on "Approve". Even approve then decline

Reply:
We are not changing any state in PO while accepting or declining the rfq. We will be showing message every time the user clicks while accepting and declining.
And we won't be hiding the accept/decline button unless the PO is confirmed.

Reply:
@vrpa-odoo So if the user clicks 50 times on decline button, the message "*The RFQ has been declined by the partner*" will be added to the chatter 50 times ? Doesn't seem a great idea. Maybe we should avoid to make a call if one already done, even if we don't refresh the view